native/core/src/parquet/parquet_support.rs

```diff
@@ -199,38 +202,147 @@ fn cast_struct_to_struct(
+        let local_file_system_url = "file:///comet/spark-warehouse/part-00000.snappy.parquet";
+        let s3_url = "s3a://test_bucket/comet/spark-warehouse/part-00000.snappy.parquet";
+        let hdfs_url = "hdfs://localhost:8020/comet/spark-warehouse/part-00000.snappy.parquet";
+
+        let all_urls = [local_file_system_url, s3_url, hdfs_url];
+        let expected: Vec<Result<(ObjectStoreUrl, Path), ExecutionError>> = vec![
+            Ok((
+                ObjectStoreUrl::parse("file://").unwrap(),
+                Path::from("/comet/spark-warehouse/part-00000.snappy.parquet"),
+            )),
+            Ok((
+                ObjectStoreUrl::parse("s3://test_bucket").unwrap(),
+                Path::from("/comet/spark-warehouse/part-00000.snappy.parquet"),
+            )),
+            Err(ExecutionError::GeneralError(
+                "Generic HadoopFileSystem error: Hdfs support is not enabled in this build"
+                    .parse()
+                    .unwrap(),
+            )),
+        ];
 
-        return Ok(url);
+        for (i, url_str) in all_urls.iter().enumerate() {
+            let url = &Url::parse(url_str).unwrap();
+            let res = prepare_object_store(Arc::new(RuntimeEnv::default()), url.to_string());
+
+            let expected = expected.get(i).unwrap();
+            match expected {
+                Ok((o, p)) => {
+                    let (r_o, r_p) = res.unwrap();
+                    assert_eq!(r_o, *o);
+                    assert_eq!(r_p, *p);
+                }
+                Err(e) => {
+                    assert!(res.is_err());
+                    let Err(res_e) = res else {
+                        panic!("test failed")
+                    };
+                    assert_eq!(e.to_string(), res_e.to_string())
+                }
+            }
+        }
     }
 
-    Err(ExecutionError::GeneralError(format!(
-        "HDFS object store cannot be created for {}",
-        url
-    )))
+    #[test]
+    #[cfg(feature = "hdfs")]
+    fn test_prepare_object_store() {
+        let hdfs_url = "hdfs://localhost:8020/comet/spark-warehouse/part-00000.snappy.parquet";
+        let expected: Result<(ObjectStoreUrl, Path), ExecutionError> = Ok((
+            ObjectStoreUrl::parse("hdfs://localhost:8020").unwrap(),
+            Path::from("/comet/spark-warehouse/part-00000.snappy.parquet"),
+        ));
+
+        let url = &Url::parse(hdfs_url).unwrap();
+        let res = prepare_object_store(Arc::new(RuntimeEnv::default()), url.to_string());
+
+        match expected {
+            Ok((o, p)) => {
+                let (r_o, r_p) = res.unwrap();
+                assert_eq!(r_o, *o);
+                assert_eq!(r_p, *p);
+            }
+            Err(e) => {
+                assert!(res.is_err());
+                let Err(res_e) = res else {
+                    panic!("test failed")
+                };
+                assert_eq!(e.to_string(), res_e.to_string())
+            }
```

Comment:
Can this match arm ever be invoked? `expected` seems to always be `Ok`

Reply:
As part of addressing this I discovered that this test was not actually being run. The test as it was written required an hdfs namenode to be running, so now I've modified it to use a `file:///` scheme which does not require the namenode. (@comphead, fyi).